cogs/SquadQueue.py

```diff
@@ -416,21 +454,40 @@ async def list_task(self):
                 await self.delete_list_messages(0)
                 return
             all_confirmed_players = mogi.players_on_confirmed_teams()
-            first_late_player_index = (mogi.num_players // mogi.players_per_room) * mogi.players_per_room
-            on_time_players = sorted(all_confirmed_players[:first_late_player_index], reverse=True)
-            late_players = all_confirmed_players[first_late_player_index:]
+            first_late_player_index = (
+                mogi.num_players // mogi.players_per_room) * mogi.players_per_room
+            if common.SERVER is common.Server.MK8DX:
+                for i, player in enumerate(all_confirmed_players, 1):
+                    if i > first_late_player_index:
+                        player.is_late = True
+                    else:
+                        player.is_late = False
 
             msg = f"**Queue closing: {discord.utils.format_dt(mogi.start_time)}**\n\n"
             msg += "**Current Mogi List:**\n"
-            for i, player in enumerate(on_time_players, 1):
-                msg += f"{i}) {player.lounge_name} ({player.mmr} MMR)\n"
-                if i % mogi.players_per_room == 0:
-                    msg += "ㅤ\n"
-            if len(on_time_players) == 0: # Text looks better this way
-                msg += "\n"
-            msg += "**Late Players:**\n"
-            for i, player in enumerate(late_players, 1):
-                msg += f"{i}) {player.lounge_name} ({player.mmr} MMR)\n"
+            if common.SERVER is common.Server.MKW:
+                on_time_players = sorted(
+                    all_confirmed_players[:first_late_player_index], reverse=True)
+                late_players = all_confirmed_players[first_late_player_index:]
+                for i, player in enumerate(on_time_players, 1):
+                    msg += f"{i}) {player.lounge_name} ({player.mmr} MMR)\n"
+                    if i % mogi.players_per_room == 0:
+                        msg += "ㅤ\n"
+                if len(on_time_players) == 0:  # Text looks better this way
+                    msg += "\n"
+                msg += "**Late Players:**\n"
+                for i, player in enumerate(late_players, 1):
+                    msg += f"{i}) {player.lounge_name} ({player.mmr} MMR)\n"
+            elif common.SERVER is common.Server.MK8DX:
+                all_confirmed_players.sort(reverse=True)
+                for i, player in enumerate(all_confirmed_players, 1):
+                    msg += f"{i}) {player.lounge_name} ({player.mmr} MMR)"
+                    if player.is_late:
+                        msg += " (late)\n"
+                    else:
+                        msg += "\n"
+                    if i % mogi.players_per_room == 0:
+                        msg += "ㅤ\n"
```

Comment:
im printing the list in mk8 as 1 big list with everyone in it, i dont know how i would know whos late after the sort by mmr unless i mark them as late first.  The mkwii method is still doing exactly what you had before

Reply:
You’re right. I reviewed this too quickly. Nothing MKW related changed.

I think MK8Dx display is convoluted still. What about this instead?
```
first_late_player_index = (mogi.num_players // mogi.players_per_room) * mogi.players_per_room
on_time_players = sorted(all_confirmed_players[:first_late_player_index], reverse=True)
late_players = all_confirmed_players[first_late_player_index:]
msg = f"**Queue closing: {discord.utils.format_dt(mogi.start_time)}**\n\n**Current Mogi List:**\n”
i = 1
# Display on time players
for i, player in enumerate(on_time_players, 1):
    msg += f"{i}) {player.lounge_name} ({player.mmr} MMR)\n"
    if i % mogi.players_per_room == 0:
        msg += "ㅤ\n"
# Display late players
for j, player in enumerate(all_confirmed_players, i):
                    msg += f"{j}) {player.lounge_name} ({player.mmr} MMR) (late)\n"
```

Written on mobile phone. Have not tested.

Reply:
Per discussion on Discord, my suggested code is incorrect for what MK8DX wants to display. Instead, consider:
```py
first_late_player_index = (mogi.num_players // mogi.players_per_room) * mogi.players_per_room
on_time_players = all_confirmed_players[:first_late_player_index]
late_players = all_confirmed_players[first_late_player_index:]
all_confirmed_players.sort(reverse=True)
msg = f"**Queue closing: {discord.utils.format_dt(mogi.start_time)}**\n\n**Current Mogi List:**\n"
# Display all on time and late players
for i, player in enumerate(all_confirmed_players, 1):
    # It's okay to lookup in late players. Since this list is guaranteed to be no more than 11 players,
    # this is a O(1) lookup time.
    late_str = " (late)" if player in late_players else ""
    msg += f"{i}) {player.lounge_name} ({player.mmr} MMR){late_str}\n"
    if i % mogi.players_per_room == 0:
        msg += "\n"
```

This code has not been tested.

Reply:
[https://github.com/emilp-595/QueueBot/pull/15/commits/2f65c81c3033b4ea9327c66acecf9fb285016234](https://github.com/emilp-595/QueueBot/pull/15/commits/2f65c81c3033b4ea9327c66acecf9fb285016234)